### PR TITLE
Fix bugs in check artifacts size and add failure conditions

### PR DESCRIPTION
**Pull request by haozheng-cobalt**
Download artifacts from all pages, by default listWorkflowRunArtifacts only downloads 1st page.
Strip off directories when matching downloaded artifacts.
Add failure conditions.

b/291954423

### CODE REVIEW

File: .github/actions/check_artifact_size/action.yaml [hunk 1]
```diff
@@ -89,16 +103,15 @@ runs:
             for (let file in fileSizeThresholds) {
               console.log(`Checking file size of ${file}.`);
 
-              const downloadFilePath = path.join('${{ steps.download-artifact.outputs.downloadDir }}', file);
+              const fileName = file.split('/').pop();
+              const downloadFilePath = path.join('${{ steps.download-artifact.outputs.downloadDir }}', fileName);
               if (!fs.existsSync(downloadFilePath)) {
-                console.error(`File ${file} was not uploaded to the main branch.`);
-                continue;
+                core.setFailed(`File ${file} was not uploaded to the main branch.`);
```
Comment (oxve) [inline .github/actions/check_artifact_size/action.yaml:109]: Add a `return;` or keep the `continue;` after this line.

File: .github/actions/check_artifact_size/action.yaml [hunk 2]
```diff
@@ -89,16 +103,15 @@ runs:
...
               }
 
               const filePath = path.join(process.env.GITHUB_WORKSPACE, '${{ inputs.path }}', file);
               if (!fs.existsSync(filePath)) {
-                console.error(`File ${filePath} was not created in the current workflow run.`);
-                continue;
+                core.setFailed(`File ${file} was not created in the current workflow run.`);
```
Comment (oxve) [inline .github/actions/check_artifact_size/action.yaml:114]: Same here.

File: .github/actions/check_artifact_size/action.yaml [hunk 3]
```diff
@@ -115,8 +128,8 @@ runs:
                 console.log(`Artifact size is decreased by ${Math.abs(deltaSize)} (${deltaThreshold}%).`);
               } else {
                 console.log(`Artifact size is increased by ${deltaSize} (${deltaThreshold}%).`);
-                if (deltaThreshold > fileSizeThresholds[file]) {
-                  const threshold = (fileSizeThresholds[file] * 100).toFixed(4);
+                const threshold = (fileSizeThresholds[file] * 100).toFixed(4);
```
Comment (oxve) [inline .github/actions/check_artifact_size/action.yaml:131]: Nit: I think the code would be a little cleaner/clearer if you didn't use `toFixed` on `deltaThreshold` and `threshold` before comparing them and only do it when printing the percentage below. Then you can also change it to `.toFixed(1)` for simplicity.

**Review by oxve — APPROVED** [review]
A couple of minor things

File: .github/actions/check_artifact_size/action.yaml (hunk 1 above)
Comment (haozheng-cobalt) [reply .github/actions/check_artifact_size/action.yaml:109]: core.setFailed will throw exception, we will not need a return; or continue;.

File: .github/actions/check_artifact_size/action.yaml (hunk 3 above)
Comment (haozheng-cobalt) [reply .github/actions/check_artifact_size/action.yaml]: yea, but that way I'll need to repeat toFixed in both if conditions.
I'll keep it as it is and change it to .toFixed(1) for simplicity.

**Pull request merged**